### PR TITLE
wiki index sorting bug and styling

### LINKED ISSUE #5317: Wiki index ordering
Our wiki page (https://publiclab.org/wiki) has various sorting options but not all of them work well. So, we need help with fixing this page.

Also, I think we can do some fix to the design of the page too, not major changes but some minor fixes like controlling width. etc so the column headings don
t take another row.

Screenshot from 2019-03-31 13-42-00

Only edits and likes sorting is working fine and other options need to be fixed.

Here's the controller method: https://github.com/publiclab/plots2/blob/master/app/controllers/wiki_controller.rb#L299-L324

And, here the view file: https://github.com/publiclab/plots2/blob/master/app/views/wiki/index.html.erb

### PR DESCRIPTION
**Pull request by GettyOrawo**
Fixes #5317 

all-sorted

**Managed**
- wikis can now be sorted by any of the columns.

**Peculiar**
- last-edited column sorts perfectly locally but doesn't seem to work in production. So I can't reproduce the bug for sorting by last-edited.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

Thanks!

### CODE REVIEW

**Comment by GettyOrawo** [comment]
`@publiclab/reviewers`  once the tests pass would love your reviews on this please. Thank you.

**Comment by GettyOrawo** [comment]
@jywarren, @gauravano the PR is ready for reviews though I can't seem to reproduce the bug for sorting by last-edited locally. It works fine on my end yet I can't tell why it breaks in production...

**Comment by gauravano** [comment]
Hey @GettyOrawo, I am getting unsorted result for page views 

Peek 2019-04-03 17-48

Checking code now..

File: app/controllers/wiki_controller.rb [hunk 1]
```diff
@@ -301,15 +301,16 @@ def index
...
+    when 'last_edited'
       order_string = 'node_revisions.timestamp DESC'
-    elsif sort_param == 'edits'
+    when 'edits'
       order_string = 'drupal_node_revisions_count DESC'
-    elsif sort_param == 'page_views'
-      order_string = 'views DESC'
-    elsif sort_param == 'likes'
+    when 'page_views'
+      order_string = 'node.vid DESC'
```
Comment (gauravano) [inline app/controllers/wiki_controller.rb]: I think here's the issue.

Comment (GettyOrawo) [reply app/controllers/wiki_controller.rb]: Thanks @gauravano , the GIF is broken so I can't see the behavior but let me give this a second look.

**Comment by GettyOrawo** [comment]
Hey @gauravano can you try now? Seems to be working okay on my local.

**Comment by GettyOrawo** [comment]
Sorry, is this still pending?

**Comment by jywarren** [comment]
Hi all! This also looks good to me, but what is the default sorting that we fall back to? I wonder if plain /wiki is getting sorted oddly due to whatever the default sorting is? 

https://publiclab.org/wiki

But I defer to @gauravano on this one. Thank you so much!!!

**Comment by jywarren** [comment]
And many apologies for slow response on this one! We appreciate your help!!!!

**Comment by alonpeer** [comment]
The default is `node_revisions.timestamp` which is `?order=last_edit`. It makes sense to me, since it also includes recently created page, so new pages will float to the top.

**Comment by gauravano** [comment]
It's working good in development but in production, many sorting options don't work right :confused:. Try on https://unstable.publiclab.org/wiki?sort=last_edited 

@jywarren @alonpeer @GettyOrawo thoughts?

**Comment by alonpeer** [comment]
Can you please tell me a bit more about the `unstable.publiclab.org` environment? Is it using the same database as production, or is it another database with the same schema but test data, or is it maybe another database with a copy of production data?
It would be much easier for me to debug this if I could have a dump of the production DB to load locally.

**Comment by GettyOrawo** [comment]
Yes @alonpeer and @gauravano  I would also find it a lot easier to test this on unstable environment. How do I push to unstable? And do I need to be authorized.. I once tried and failed for not having rights.

**Comment by gauravano** [comment]
> Can you please tell me a bit more about the unstable.publiclab.org environment?

We use unstable to test changes. Only some people have push access but if you want your PR to be test there, you can ask in Gitter channel and someone else can do it for you.

> Is it using the same database as production, or is it another database with the same schema but test data, or is it maybe another database with a copy of production data?

It's another database with a copy from production data.
 
> It would be much easier for me to debug this if I could have a dump of the production DB to load locally.

I think, it's not possible. Although, @icarito can better answer that.

**Comment by SidharthBansal** [comment]
How is it going here?

**Comment by GettyOrawo** [comment]
hey @SidharthBansal  this is one of those "It works on my machine and not in production" kind of bugs. Didn't try pushing to unstable though.

**Comment by SidharthBansal** [comment]
Please resolve the conflicts and try to push it on unstable

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR OR reopen this PR and add additional changes to it.
Thanks for contributing on Public Lab

**Pull request closed without merging**